Library/Homebrew/tap.rb

```diff
@@ -25,12 +27,22 @@ class Tap
   # e.g. `/usr/local/Library/Taps/user/homebrew-repo`
   attr_reader :path
 
+  # The path of configuration for this {Tap}
+  # e.g. 'usr/local/Library/Taps/user/repo.json'
+  attr_reader :json_path
+
+  # If this {Tap} is a pinned one
+  # Pinned taps will be prioritized over core if formulae names are supplied by the user
+  attr_reader :pinned
+
   def initialize(user, repo)
     # we special case homebrew so users don't have to shift in a terminal
     @user = user == "homebrew" ? "Homebrew" : user
     @repo = repo
     @name = "#{@user}/#{@repo}".downcase
     @path = TAP_DIRECTORY/"#{@user}/homebrew-#{@repo}".downcase
+    @json_path = TAP_DIRECTORY/"#{@user}/#{@repo}.json".downcase
+    @pinned = @json_path.exist?? Utils::JSON.load(File.read(@json_path))["pinned"] : false
```

Comment:
I see. It might cause performance issue when there are many taps tapped. I am going to make the modification and run a benchmark with a lot of taps installed soon.